SecondHomework/Logistic/FileManager.cs

```diff
@@ -2,25 +2,67 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Globalization;
 
 public class FileManager
 {
     private const string _filePath = @"Parcels.json";
+    private const string _deliveredParcelsFilePath = @"DeliveredParcels.json";
+    private static readonly SemaphoreSlim _deliverySemaphore = new SemaphoreSlim(2, 2);
 
-    public void Save(List<Parcel> parcels)
+    public async Task SaveParcelsAsync(List<Parcel> parcels)
     {
         var json = JsonConvert.SerializeObject(parcels, Newtonsoft.Json.Formatting.Indented);
-        File.WriteAllText(_filePath, json);
+        await File.WriteAllTextAsync(_filePath, json);
     }
 
-    public List<Parcel> Read()
+    public async Task<List<Parcel>> ReadAsync()
     {
+
         if (!File.Exists(_filePath))
         {
-            Save(new List<Parcel>());
+            return new List<Parcel>();
         }
-        var json = File.ReadAllText(_filePath);
+        var json = await File.ReadAllTextAsync(_filePath);
         List<Parcel> parcelsList = JsonConvert.DeserializeObject<List<Parcel>>(json) ?? new List<Parcel>();
         return parcelsList;
     }
+
+    public async Task<List<Parcel>> ReadDeliveredAsync()
+    {
+        await _deliverySemaphore.WaitAsync(); // Захист доступу до файлу DeliveredParcels.json
+        try
+        {
+            if (!File.Exists(_deliveredParcelsFilePath))
+            {
+                // Якщо файл не існує, створюємо новий файл з пустим списком
+                await File.WriteAllTextAsync(_deliveredParcelsFilePath, "[]");
```

Comment:
You are breaking S principle here, I would create file on write, not on read. Here you can just return empty collection